### PR TITLE
Use webpack to build client side assets

### PR DESCRIPTION
**Pull request by deviozc**
- Manage dependencies using import
- Build dependencies into vendor.js file.
- build app related js to app.js

### CODE REVIEW

**Comment by deviozc** [comment]
related to #123

File: covasim/webapp/index.html [hunk 1]
```diff
@@ -12,20 +12,19 @@
   <link type="text/css" rel="stylesheet" href="https://unpkg.com/bootstrap-vue@2.11.0/dist/bootstrap-vue.min.css"/>
   <link type="text/css" rel="stylesheet" href="assets/themify-icons.css"/>
```
Comment (lgeorge-idm) [inline covasim/webapp/index.html:13]: Do we still need this if you have the vendor.css?

File: covasim/webapp/package.json [hunk 2]
```diff
@@ -4,15 +4,30 @@
...
+    "build:prod": "webpack --config webpack.config.js --mode=production"
+  },
   "author": "Institute for Disease Modeling",
   "license": "CC-BY-SA-4.0",
   "dependencies": {
-    "bootstrap": "4.4.1",
+    "bootstrap": "^4.4.1",
     "bootstrap-vue": "2.11.0",
     "file-saver": "2.0.2",
-    "plotly.js-dist": "1.53.0",
```
Comment (lgeorge-idm) [inline covasim/webapp/package.json:14]: Please add this back. It's needed for dependency tracking.

File: covasim/webapp/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        new MiniCssExtractPlugin()
+    ],
+
+    output: {
+        filename: '[name].js',
+        path: path.resolve(__dirname, 'dist'),
+    },
+    externals: {
+        "plotly": "Plotly"
+    }
```
Comment (lgeorge-idm) [inline covasim/webapp/webpack.config.js:62]: Why was Plotly left as an external?

File: covasim/webapp/index.html (hunk 1 above)
Comment (lgeorge-idm) [reply covasim/webapp/index.html:13]: > Do we still need bootstrap-vue.css if you have the vendor.css?

File: covasim/webapp/package.json (hunk 2 above)
Comment (deviozc) [reply covasim/webapp/package.json:14]: Updated to use plotly.js-cartesian-dist-min for smaller bundle size

File: covasim/webapp/index.html (hunk 1 above)
Comment (deviozc) [reply covasim/webapp/index.html]: Removed. It is now moved to vendor css.

File: covasim/webapp/package.json (hunk 2 above)
Comment (deviozc) [reply covasim/webapp/package.json:14]: Updated to use plotly.js-cartesian-dist-min for smaller bundle size

File: covasim/webapp/webpack.config.js (hunk 3 above)
Comment (deviozc) [reply covasim/webapp/webpack.config.js]: Plotly js was quite big, it was 3mb minified js.
I switched to plotly.js-cartesian-dist-min which is about 900kb.

**Comment by cliffckerr** [comment]
@lgeorge-idm @deviozc I was wrong, you were right, let's use webpack :) I'll try to resolve the merge conflicts with this PR

**Comment by lgeorge-idm** [comment]
@cliffckerr , I can take on webpackifying.

**Comment by cliffckerr** [comment]
Stop press! Let's not use webpack :rofl:

**Pull request closed without merging**